packages/cli/src/controllers/tags.controller.ts

```diff
@@ -1,54 +1,72 @@
+import { UpdateTagRequestDto, CreateTagRequestDto } from '@n8n/api-types';
+import { RetrieveTagQueryDto } from '@n8n/api-types/src/dto/tag/retrieve-tag-query.dto';
+import { GlobalConfig } from '@n8n/config';
 import { Request, Response, NextFunction } from 'express';
 
-import config from '@/config';
-import { Delete, Get, Middleware, Patch, Post, RestController, GlobalScope } from '@/decorators';
-import { BadRequestError } from '@/errors/response-errors/bad-request.error';
-import { TagsRequest } from '@/requests';
+import {
+	Delete,
+	Get,
+	Middleware,
+	Patch,
+	Post,
+	RestController,
+	GlobalScope,
+	Body,
+	Param,
+	Query,
+} from '@/decorators';
+import { AuthenticatedRequest } from '@/requests';
 import { TagService } from '@/services/tag.service';
 
 @RestController('/tags')
 export class TagsController {
-	private config = config;
-
-	constructor(private readonly tagService: TagService) {}
+	constructor(
+		private readonly tagService: TagService,
+		private readonly globalConfig: GlobalConfig,
+	) {}
 
 	// TODO: move this into a new decorator `@IfEnabled('workflowTagsDisabled')`
 	@Middleware()
-	workflowsEnabledMiddleware(_req: Request, _res: Response, next: NextFunction) {
-		if (this.config.getEnv('workflowTagsDisabled'))
-			throw new BadRequestError('Workflow tags are disabled');
+	workflowsEnabledMiddleware(_req: Request, res: Response, next: NextFunction) {
+		if (this.globalConfig.tags.workflowTagsDisabled) {
+			res.status(400).json({ message: 'Workflow tags are disabled' });
+			return;
+		}
 		next();
 	}
 
 	@Get('/')
 	@GlobalScope('tag:list')
-	async getAll(req: TagsRequest.GetAll) {
-		return await this.tagService.getAll({ withUsageCount: req.query.withUsageCount === 'true' });
+	async getAll(_req: AuthenticatedRequest, _res: Response, @Query query: RetrieveTagQueryDto) {
```

Comment:
once we have migrated all controllers, we can remove these implicit parameters 🤞🏽